### PR TITLE
Add alternate basis fake cambridge backend

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

To test that the full execute path works correctly with a real backend
using an alternative basis set this commit adds a new fake backend
FakeCambridgeAlternativeBasis which is identical to FakeCambridge except
that the basis gates are overriden from u1/u2/u3 to be p/sx/u and
alternative properties file json was manually created by copying the
existing one for FakeCambridge and replacing all instances of u1
and u3 to p and u respectively.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
This is failing tests because released aer (0.6.1) doesn't understand p/sx/u. Phase and sqrt x were added to aer for the pending 0.7.0 release in https://github.com/Qiskit/qiskit-aer/pull/951 but it looks like it doesn't understand u: https://github.com/Qiskit/qiskit-aer/blob/master/qiskit/providers/aer/backends/qasm_simulator.py#L249-L255 FWIW, this worked locally with basicaer, but we run with aer in CI for some jobs to test the fake backends with noise.

**Comment by mtreinish** [comment]
Opened an issue for the missing u gate support on aer: https://github.com/Qiskit/qiskit-aer/issues/988

File: qiskit/test/mock/backends/cambridge/fake_cambridge.py [hunk 1]
```diff
@@ -56,3 +60,19 @@ def properties(self):
                 props = json.load(f_prop)
             self._properties = BackendProperties.from_dict(props)
         return self._properties
+
+
+class FakeCambridgeAlternativeBasis(FakeCambridge):
```
Comment (kdk) [inline qiskit/test/mock/backends/cambridge/fake_cambridge.py:65]:
```suggestion
class FakeCambridgeAlternativeBasis(FakeCambridge):
    """A fake Cambridge backend with alternate 1q basis gates."""
```

**Review by kdk — COMMENTED** [review]
Looks good. One lint error.

**Review by kdk — APPROVED** [review]

**Pull request merged**